File: PHP/6php 1/find_missing_letter.php
<?php
/*
|--------------------------------------------------------------------------
| Find the missing letter
|--------------------------------------------------------------------------
|
*/

function find_missing_letter($array) {
   for ($i = 0; $i < count($array) - 1; $i++) {
      if (ord($array[$i]) + 1 !== ord($array[$i + 1])) {
         return chr(ord($array[$i]) + 1);
      }
   }
}


function find_missing_letter(array $a): string {
   return join(array_diff(range($a[0],end($a)),$a));
}


function find_missing_letter(array $array): string {
   $letter = range(reset($array),end($array));
   return current(array_diff($letter, $array));
}


print_r(find_missing_letter(["a", "b", "c", "d", "f"]), "e");
print_r(find_missing_letter(["O", "Q", "R", "S"]), "P");

File: Python/6py 2/find_missing_letter.py
"""
|--------------------------------------------------------------------------
| Find the missing letter
|--------------------------------------------------------------------------
|
"""

def find_missing_letter(chars):
   for i in range(len(chars) - 1):
      if ord(chars[i]) + 1 != ord(chars[i + 1]):
         return chr(ord(chars[i]) + 1)
      

def find_missing_letter(c):
   return next(chr(ord(c[i])+1) for i in range(len(c)-1) if ord(c[i])+1 != ord(c[i+1]))
      

def find_missing_letter(chars):
   n = 0
   while ord(chars[n]) == ord(chars[n+1]) - 1: n += 1
   return chr(1+ord(chars[n]))
      

print(find_missing_letter(['a','b','c','d','f']), 'e')
print(find_missing_letter(['O','Q','R','S']), 'P')
print(find_missing_letter(['b','d']), 'c')